tests/suites/test_suite_aes.function

```diff
@@ -418,7 +418,7 @@ exit:
 /* END_CASE */
 
 /* BEGIN_CASE */
-void aes_invalid_mode()
+void aes_invalid_mode(void)
```

Comment:
I'd prefer to keep allowing `()` in test functions. It's a more logical syntax (which C lacks only due to backward compatibility constraint), and if we stop allowing it, we'll likely forget in new work and then get surprised when the CI fails with `clang-latest`. If we do, then we don't need the massive (in terms of changed lines) changes to `*.function`.

This only requires a small change in `tests/scripts/generate_test_code.py`: in `parse_function_code`, if `args` is blank (ideally also if `args` only contains comments, but that's not critical), add `void` to it.

Reply:
Makes sense. Will drop the `*.function` changes.

Reply:
There is a condition in `tests/scripts/generate_test_code.py` in function `gen_from_test_data` 
`if len(test_args) != len(func_args):`
which checks if the test data and the function signature have the same number of arguments. 
This reports an error saying the function generated and the function parameter listed in `*.data` are not the same if I add `void` insertion in the script. :( 
Do you want me to update the script and all the test data files which use the void arg functions? :)

Reply:
Either way, we will end up changing lots of files :)

Reply:
There's already code in the script to accept `void` as meaning no arguments, since 47e2e8817d938bc02ec94be8f01a4755eacca4c5. Looking at it now (I'd forgotten how it was done), it seems that we're stripping `void` if it was present in the source, so we should unconditionally add it when generating. So yes, adding it in `gen_from_test_data` does sound good.

Reply:
By the way, there are unit tests in `tests/scripts/test_generate_test_code.py`. If inserting `void` breaks a unit test, please update the expected result from that test. If inserting `void` doesn't break an existing unit test, please add one.